internal/highway/x/ipfs/protocol.go

```diff
@@ -49,13 +52,32 @@ func DecodeCIDFromString(s string) (cid.Cid, error) {
 }
 
 // GetData returns a file from IPFS.
-func (i *IPFSProtocol) GetData(cid string) ([]byte, error) {
+func (i *Protocol) GetData(cid string) ([]byte, error) {
 	// Decode CID from String
 	c, err := DecodeCIDFromString(cid)
 	if err != nil {
 		return nil, err
 	}
 
+	ctx := context.TODO()
```

Comment:
Use `context.Background()` for a new context

Reply:
The reason I used TODO here is because I assumed the context should be coming from outside, and be passed to the protocol functions. I will make it background now, but I think we should rethink these.